src/v2/view-builder/internals/BaseOktaVerifyChallengeView.js

```diff
@@ -153,10 +154,10 @@ const Body = BaseFormWithPolling.extend({
               }
             });
         })
-        .fail(onFailure);
+        .catch(onFailure);
     };
 
-    let probeChain = Promise.resolve();
+    let probeChain = $.Deferred().resolve();
```

Comment:
will this deferred cause any degradation of UX like slower authentication speed?

Reply:
With this change the behaviour will be same as in Gen3 - probe/challenge calls will be running not in parallel, but in chain (probing/challenging for port 2 should wait for probing/challenging of port 1 to complete). 
This can be slower.
But such behaviour in expected judging from code and tests.

If it's desired to have parallel requests, then tests should be rewritten to reflect this.
For example, this line should be removed:
https://github.com/okta/okta-signin-widget/blob/c07d759f6e0e57ea6eecb690e1ae696c4bf3aa96/test/testcafe/spec/DeviceChallengePollView_spec.js#L430

Reply:
cc @shuowu-okta @Luxu2-Okta

Reply:
After talking with the team, we would like to fix it, to make V2 behave similar like V3, there shouldn't be a performance concern.
We can be more cautious by using a new flag from server, but this might be an overkill

Reply:
We typically isolate higher-risk changes in a patch bump on the tip so it can be easily rolled back. I think this can target `7.22.1`

Reply:
How does using `$.Deferred().resolve()` differ from `Promise.resolve()` and how does it cause the probe chain to become serial?

Reply:
the `$.Deferred().resolve()` is the jquery's promise, not sure about the very low level, but jquery promise has the `.done` method (similar as `.then`), while es6 promise doesn't.
since all the probing will use jquery $.ajax method, so I guess using `$.Deferred().resolve()` and return the rejected promise in `onFailure` will make it serial, but I will leave it to @denysoblohin-okta  to confirm

Reply:
> How does using $.Deferred().resolve() differ from Promise.resolve()

First one returns `$.Deferred` object instead of `Promise`. This will change `probeChain` from a chain of Promises to a chain of Deferred. `Deferred` is a "thenable" object, but not a real promise.


> how does it cause the probe chain to become serial?

In current code native Promises are mixed with jQuery's Deferred:
https://github.com/okta/okta-signin-widget/blob/f37fdb316d818e46f9f93349b3b534d84ebf6831/src/v2/view-builder/internals/BaseOktaVerifyChallengeView.js#L166
https://github.com/okta/okta-signin-widget/blob/f37fdb316d818e46f9f93349b3b534d84ebf6831/src/v2/view-builder/internals/BaseOktaVerifyChallengeView.js#L22
(`$.ajax` returns `$.Deferred`)

`probeChain` uses native Promises with `then/catch` API. But `doProbing()` returns `$.Deferred` and uses `done/fail` API.
This is a problem. To achieve proper chaining we need to change `done/fail` API to `then/catch`.


Actually we can leave `Promise.resolve()` instead of `$.Deferred().resolve()`, then we still have mixing of Promise and Deferred, but they can play well together with `then/catch` API.

However there was a concern from @shuowu-okta to use Promises in Gen2 (when my previous fix contained promisification of `$.ajax`, to use Promises instead of Deferred)

Reply:
Rebased PR onto `7.21`

Reply:
just curious, this will target `7.21.2` in the 2024.09.0?

Reply:
> Actually we can leave Promise.resolve() instead of $.Deferred().resolve()

Yeah, I think now that the `Promise` API is standardized and canonically polyfilled we should try to use `Promise` when we can since it's better understood and well-documented (MDN).